imod/mf6/model.py

```diff
@@ -542,17 +546,18 @@ def _get_regridding_domain(
             idomain, target_grid=target_grid
         )
         included_in_all = None
-        for regriddertype, function in methods.items():
-            regridder = regridder_collection.get_regridder(
-                regriddertype,
-                function,
-            )
-            regridded_idomain = regridder.regrid(idomain)
-            if included_in_all is None:
-                included_in_all = regridded_idomain
-            else:
-                included_in_all = included_in_all.where(regridded_idomain.notnull())
-
+        for regriddertype, functionlist in methods.items():
+            for function in functionlist:
+                regridder = regridder_collection.get_regridder(
+                    regriddertype,
+                    function,
+                )
+                regridded_idomain = regridder.regrid(idomain)
+                if included_in_all is None:
+                    included_in_all = regridded_idomain
+                else:
+                    included_in_all = included_in_all.where(regridded_idomain.notnull())
+                    included_in_all = regridded_idomain.where(regridded_idomain <=0, other = included_in_all)
```

Comment:
I haven't tested this snippet but could the following work?
```
regridders = [regridder_collection.get_regridder(regriddertype,function) for function in functionlist for regriddertype, functionlist in methods.items()]
included_in_all = [regridder.regrid(idomain) for regridder in regridders if regridder.regrid(idomain).notnull()]
```
or 
```
regridders = [regridder_collection.get_regridder(regriddertype,function) for function in functionlist for regriddertype, functionlist in methods.items()]
regridded_domains = [regridder.regrid(idomain) for regridder in regridders]
included_in_all = [regridded_domain for regridded_domain in regridded_domains if regridded_domain.notnull()]
```

Reply:
not done. The first statement produces strange behavior:
to test why in the debugger I first tried a simpler expression:
`[str(functionlist) for regriddertype, functionlist in methods.items()]`
this produces : `["['mean', ..., 'mode']"]`
But then adding the second loop fails":
`[ str(function) for function in functionlist for regriddertype, functionlist in methods.items()]`
produces:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'functionlist' is not defined
```
Also,- the final statement in the suggestion makes included_in_all into a list which is not really what we want.

Reply:
Could you try to get this to work?
It would greatly simply the code. Instead of having multiple if statements in a for loop in a for loop
It also makes it clearer what you are trying to do because each of the list comprehensions has a single goal (Instead of having the for loops do multiple things)

Reply:
> But then adding the second loop fails": `[ str(function) for function in functionlist for regriddertype, > functionlist in methods.items()]` produces:
> 
> ```
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
> NameError: name 'functionlist' is not defined
> ```

For nested list comprehensions, you have to change the order into:
`[ str(function) for regriddertype, functionlist in methods.items() for function in functionlist]`

[See this explanation on how nested list comprehensions work](https://stackoverflow.com/a/20639246)

Reply:
done, now making a list of regridders. I kept the do-loop because include_in_all is not a list. Simplified the do-loop by initializing include_in_all upfront.